sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -437,6 +437,18 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
   private def putStateIntoStateCacheMap(
       newVersion: Long,
       map: HDFSBackedStateStoreMap): Unit = synchronized {
+    val loadedEntries = loadedMaps.size()
+    val lastKey: Option[Long] = if (loadedEntries > 0) Some(loadedMaps.lastKey()) else None
+    if (lastKey.isDefined) {
+      logInfo(s"Trying to add version=$newVersion to state cache map with " +
+        s"current_size=$loadedEntries and last_loaded_version=${lastKey.get} and " +
```

Comment:
maybe "earliest_loaded_version"?

Reply:
This should return last/highest key/version present in the map right ? so maybe ok to keep this as it is ?

Reply:
the keys are stored in the reverse order, so the last key is the earliest version: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala#L385-L386

Reply:
Yup, agreed. Updated the diff. Thx